File: src/services/IntentRouter.ts
import { IntentType, Route } from "../types";

export class IntentRouter {
  public static route(command: string): Route {
    const c = command.trim();
    const l = c.toLowerCase();

    // Greeting patterns (English & Hindi)
    if (/^(hello|hi|hlo|helo|hey|namaste|yo|sup)(\s+jarvis)?[.!?]*$/i.test(l)) {
      return { type: IntentType.GREETING, value: "", originalCommand: c };
    }

    // App launch
    if (l.startsWith("open ") || l.startsWith("launch ")) {
      const target = c.replace(/^(open|launch)\s+/i, "").trim();
      return { type: IntentType.OPEN_APP, value: target, originalCommand: c };
    }

    // Web search
    if (l.startsWith("search ") || l.startsWith("find ") || l.startsWith("google ")) {
      const query = c.replace(/^(search|find|google)\s+/i, "").trim();
      return { type: IntentType.SEARCH, value: query, originalCommand: c };
    }

    // Phone call
    if (l.startsWith("call ") || l.startsWith("dial ")) {
      const target = c.replace(/^(call|dial)\s+/i, "").trim();
      return { type: IntentType.CALL, value: target, originalCommand: c };
    }

    // SMS / Message
    if (l.startsWith("message ") || l.startsWith("sms ") || l.startsWith("text ")) {
      const target = c.replace(/^(message|sms|text)\s+/i, "").trim();
      return { type: IntentType.MESSAGE, value: target, originalCommand: c };
    }

    // Settings
    if (l.includes("settings")) {
      return { type: IntentType.SETTINGS, value: "", originalCommand: c };
    }

    // Focus timer
    if (l.includes("timer") || /.*\d+\s*(minute|min|m|second|sec|s).*/i.test(l)) {
      return { type: IntentType.TIMER, value: c, originalCommand: c };
    }

    // Default to conversational Gemini AI
    return { type: IntentType.CHAT, value: c, originalCommand: c };
  }
}

File: app/src/main/java/com/ayush9182/jarvis/IntentRouter.kt
package com.ayush9182.jarvis

class IntentRouter {
    enum class IntentType { GREETING, OPEN_APP, SEARCH, CALL, MESSAGE, SETTINGS, TIMER, CHAT }
    data class Route(val type: IntentType, val value: String = "")

    fun route(command: String): Route {
        val c = command.trim()
        val l = c.lowercase()
        return when {
            l.matches(Regex("^(hello|hi|hlo|helo|hey|namaste|yo|sup)(\\s+jarvis)?[.!?]*$")) -> Route(IntentType.GREETING)
            l.startsWith("open ") || l.startsWith("launch ") -> Route(IntentType.OPEN_APP, c.substringAfter(' '))
            l.startsWith("search ") || l.startsWith("find ") || l.startsWith("google ") -> Route(IntentType.SEARCH, c.substringAfter(' '))
            l.startsWith("call ") || l.startsWith("dial ") -> Route(IntentType.CALL, c.substringAfter(' '))
            l.startsWith("message ") || l.startsWith("sms ") || l.startsWith("text ") -> Route(IntentType.MESSAGE, c.substringAfter(' '))
            l.contains("settings") -> Route(IntentType.SETTINGS)
            l.contains("timer") || l.matches(Regex(".*\\d+\\s*(minute|min|m|second|sec|s).*")) -> Route(IntentType.TIMER, c)
            else -> Route(IntentType.CHAT, c)
        }
    }
}
